terraform/aks/data.tf

```diff
@@ -0,0 +1,12 @@
+data "azurerm_key_vault" "infra_secret_vault" {
```

Comment:
Can we [use a data source](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/postgresql_flexible_server) here to avoid relying on the key vault content?

Reply:
As discussed, the data source doesn't provide the password so it can't be used here